packages/@o3r/components/src/stores/placeholder-effect/placeholder.effect.ts

```diff
@@ -34,18 +31,26 @@ export class PlaceholderTemplateResponseEffect {
       fromApiEffectSwitchMapById(
         (templateResponse, action) => {
           const facts = templateResponse.vars ? Object.entries(templateResponse.vars).filter(([, variable]) => variable.type === 'fact') : [];
-          const factsStreamsList = this.rulesEngineService ? facts.map(([varName, fact]) =>
-            this.rulesEngineService!.engine.retrieveOrCreateFactStream(fact.value).pipe(
-              map((factValue) => ({
-                varName,
-                factName: fact.value,
-                // eslint-disable-next-line new-cap
-                factValue: (fact.path && factValue) ? JSONPath({ wrap: false, json: factValue, path: fact.path }) : factValue
-              })),
-              distinctUntilChanged((previous, current) => previous.factValue === current.factValue)
-            )) : [];
-
-          const factsStreamsList$ = factsStreamsList.length ? combineLatest(factsStreamsList) : of([]);
+          const factsStreamsList$ = from(import('@o3r/rules-engine')).pipe(
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            switchMap(({ RulesEngineRunnerService }) => {
```

Comment:
With the dynamic import you'll always have a value for the service even if it is not provided in the app, but the module is installed. So I guess we are losing the 'optional' feature here.
Correct me if I am wrong.

Reply:
If "a value" is "a class reference" yes, but it does not mean that we have the class instance.
According where it is injected (if not injected inRoot), it can be present only if the module it is attached to is loaded.

Reply:
I see, thanks for the explanation :)